src/block_utils.jl

```diff
@@ -0,0 +1,57 @@
+
+# Reduced QR factorization with Householder reflections:
+# Q, R = householder(A)
+#
+# Input :
+# A an n-by-k matrix, n ≥ k
+#
+# Output :
+# Q an n-by-k orthonormal matrix: QᴴQ = Iₖ
+# R an k-by-k upper triangular matrix: QR = A
+function householder(A::AbstractMatrix{FC}) where FC <: FloatOrComplex
+  n, k = size(A)
+  Q = copy(A)
+  τ = zeros(FC, k)
+  R = zeros(FC, k, k)
+  householder!(Q, R, τ)
+end
+
+function householder!(Q::AbstractMatrix{FC}, R::AbstractMatrix{FC}, τ::AbstractVector{FC}) where FC <: FloatOrComplex
+  n, k = size(Q)
+  R .= zero(FC)
```

Comment:
I would prefer using 
```julia
fill!(R, zero(FC))
```
(my rule of thumb is to avoid using broadcasting if we can)

Reply:
Yes good idea, broadcast generates a new kernel at each iteration.